lib/block-supports/duotone.php

```diff
@@ -340,8 +311,91 @@ function ( $selector ) use ( $duotone_id ) {
 	</svg>
 
 	<?php
+	$duotone_svg = ob_get_clean();
+	return $duotone_svg;
+}
 
-	$duotone = ob_get_clean();
+/**
+ * Get Duotone colors from array.
+ *
+ * @param array $duotone_colors The array of colors that define the duotone style.
+ * @return array The array of values mapped to rgb.
+ */
+function get_duotone_color_values( $duotone_colors ) {
+	$duotone_values = array(
+		'r' => array(),
+		'g' => array(),
+		'b' => array(),
+	);
+	foreach ( $duotone_colors as $color_str ) {
+		$color = gutenberg_tinycolor_string_to_rgb( $color_str );
+
+		$duotone_values['r'][] = $color['r'] / 255;
+		$duotone_values['g'][] = $color['g'] / 255;
+		$duotone_values['b'][] = $color['b'] / 255;
+	}
+
+	return $duotone_values;
+}
+
+/**
+ * Output Duotone markup in the footer.
+ *
+ * @param $duotone_markup The markup for the SVG filer, and if necessary the style tag.
+ */
+function gutenberg_output_duotone_markup( $duotone_markup ) {
+	add_action(
+		// Ideally we should use wp_head, but SVG defs can't be put in there.
+		'wp_footer',
+		function () use ( $duotone_markup ) {
+			echo $duotone_markup;
+		}
+	);
+}
+
+/**
+ * Render out the duotone stylesheet and SVG.
+ *
+ * @param  string $block_content Rendered block content.
+ * @param  array  $block         Block object.
+ * @return string                Filtered block content.
+ */
+function gutenberg_render_duotone_support( $block_content, $block ) {
```

Comment:
Github is confused. None of this code is new, it's just broken into smaller functions.